apps/backend/src/groups/groups.service.ts

```diff
@@ -49,6 +51,18 @@ export class GroupsService {
     });
   }
 
+  async updateUserGroupRole(
+    group: Group,
+    updateGroupUser: AddUserToGroupDto
+  ): Promise<GroupUser | undefined> {
+    return group.users
+      .find((userToUpdate) => updateGroupUser.userId === userToUpdate.id)
```

Comment:
We have a user id from the DTO and we have the group id. It is more efficient to query for the GroupUser directly using SQL and not iterate a potentially large list of users in a group.